server/lease/lessor.go

```diff
@@ -572,6 +572,9 @@ func (le *lessor) Detach(id LeaseID, items []LeaseItem) error {
 
 	l := le.leaseMap[id]
 	if l == nil {
+		for _, it := range items {
+			delete(le.itemMap, it)
+		}
 		return ErrLeaseNotFound
```

Comment:
thx for review. I have a litte worry.

Reply:
Thanks @chaochn47 .  Indeed https://github.com/etcd-io/etcd/pull/14080 introduced the leak, will send a PR to rollback the change. 

Actually the deadlock fixed in https://github.com/etcd-io/etcd/pull/14080 doesn't likely happen because all lease operations (grant, revoke, checkpoint) are executed sequentially in the applying workflow. If anyone still thinks it's possible to run into a deadlock, please create a test to reproduce it firstlhy.